packages/flutter_tools/lib/src/commands/run.dart

```diff
@@ -592,6 +593,11 @@ class RunCommand extends RunCommandBase {
     if (devices == null) {
       throwToolExit(null);
     }
+
+    if (devices!.any((Device device) => device is MacOSDesignedForIPadDevice)) {
```

Comment:
Okay I see now, I led you astray, this breaks the `flutter run -d all` target that runs on all eligible devices (`-d all` doesn't work for `drive` or `attach`):
```
$ flutter run -d all
Checking for wireless devices...
Mac Designed for iPad is currently not supported for flutter run -d.
```
and then it tool exits.
However with what you had before where it just checks if one is selected, it tries and fails:
```
Launching lib/main.dart on Mac Designed for iPad in debug mode...
...
[ERROR:flutter/shell/platform/darwin/graphics/FlutterDarwinContextMetalImpeller.mm(42)] Using the Impeller rendering backend.
Installing and launching...                                        22.9s

Oops; flutter has exited unexpectedly: "UnimplementedError: Building for "Mac Designed for iPad" is not supported.".
```
```
UnimplementedError: UnimplementedError: Building for "Mac Designed for iPad" is not supported.

#0      MacOSDesignedForIPadDevice.startApp (package:flutter_tools/src/macos/macos_ipad_device.dart:74:5)
#1      FlutterDevice.runHot (package:flutter_tools/src/resident_runner.dart:465:55)
<asynchronous suspension>
#2      HotRunner.run.<anonymous closure> (package:flutter_tools/src/run_hot.dart:440:14)
<asynchronous suspension>
#3      Future.wait.<anonymous closure> (dart:async/future.dart:518:21)
<asynchronous suspension>
#4      HotRunner.run (package:flutter_tools/src/run_hot.dart:471:34)
<asynchronous suspension>
#5      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:812:27)
<asynchronous suspension>
#6      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1398:27)
<asynchronous suspension>
#7      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:153:19)
<asynchronous suspension>
#8      CommandRunner.runCommand (package:args/command_runner.dart:212:13)
<asynchronous suspension>
#9      FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:355:9)
<asynchronous suspension>
#10     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:153:19)
<asynchronous suspension>
#11     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:295:5)
<asynchronous suspension>
#12     run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:119:9)
<asynchronous suspension>
```

Instead, you could revert to the check you had before this commit where if there's only one selected and it's `MacOSDesignedForIPadDevice` then toolExit.  And then after that check `globals.deviceManager!.hasSpecifiedAllDevices` and remove the `MacOSDesignedForIPadDevice` device from the `devices` property.  Does that make sense?

Reply:
This still doesn't work, you need to first check `-d mac-designed-for-ipad` and fail with a reasonable error, do it doesn't just say "No supported devices found with name or id matching 'mac-designed-for-ipad'" or whatever it would say. So you should make this something like what you had before
```suggestion
    if (devices!.length == 1 && devices!.first is! MacOSDesignedForIPadDevice) {
```
And then the `hasSpecifiedAllDevices` case you have below will filter out the `all` scenario.
To test this `flutter create` a new project then run `flutter run -d all`.

Reply:
Ah. For some reason it did not connect in my brain that run -d all should still run on all the other devices.